### PR TITLE
nft mint wrapper

### PR DESCRIPTION
**Pull request by mikiquantum**
<This is intended as a guide when describing-reviewing pull requests for go-centrifuge>
Issue Link: <Github link>

### CODE REVIEW

File: protobufs/nft/service.proto [hunk 1]
```diff
@@ -21,12 +21,30 @@ service NFTService {
...
   string transaction_id = 5;
 }
 
+message NFTPaymentObligationRequest {
+  // Invice Document identifier
+  string identifier = 1;
+  // Deposit address for NFT Token created
+  string deposit_address = 2;
+  // The contract address of the registry where the token should be minted
+  string registry_address = 3;
```
Comment (mikiquantum) [inline protobufs/nft/service.proto]: Do we still need to provide the `registry_address` or can we read it from the config? or is it something else? @vedhavyas @lucasvo

Comment (vedhavyas) [reply protobufs/nft/service.proto]: for the nft pay-ob wrapper, we should ideally pick it from the config.

File: testworld/nft_test.go [hunk 2]
```diff
@@ -33,19 +35,24 @@ func TestPaymentObligationMint_invoice_successful(t *testing.T) {
 	for _, c := range tests {
 		t.Run("", func(t *testing.T) {
 			t.Parallel()
-			paymentObligationMint(t, typeInvoice, c.grantAccess, c.tokenProof, c.readAccessProof)
+			paymentObligationMint(t, typeInvoice, c.grantAccess, c.tokenProof, c.readAccessProof, false)
```
Comment (vimukthi-git) [inline testworld/nft_test.go]: could you change this test not to be a table test?

File: testworld/nft_test.go [hunk 3]
```diff
@@ -70,47 +77,68 @@ func paymentObligationMint(t *testing.T, documentType string, grantNFTAccess, to
...
+		keys, err := acc.GetKeys()
+		if err != nil {
+			t.Error(err)
+		}
+		signerId := hexutil.Encode(append(alice.id[:], keys[identity.KeyPurposeSigning.Name].PublicKey...))
+		signingRoot := fmt.Sprintf("%s.%s", documents.DRTreePrefix, documents.SigningRootField)
+		signatureSender := fmt.Sprintf("%s.signatures[%s].signature", documents.SignaturesTreePrefix, signerId)
+
+		// mint an NFT
+		test := struct {
```
Comment (vimukthi-git) [inline testworld/nft_test.go]: lets remove these unnecessary structs

File: testworld/nft_test.go [hunk 4]
```diff
@@ -70,47 +77,68 @@ func paymentObligationMint(t *testing.T, documentType string, grantNFTAccess, to
...
+				"proofFields":               []string{proofPrefix + ".gross_amount", proofPrefix + ".currency", proofPrefix + ".date_due", proofPrefix + ".sender", proofPrefix + ".status", signingRoot, signatureSender, documents.CDTreePrefix + ".next_version"},
+				"submitTokenProof":          tokenProof,
+				"submitNftOwnerAccessProof": nftReadAccessProof,
+				"grantNftAccess":            grantNFTAccess,
+			},
+		}
+		response, err = alice.host.mintNFT(alice.httpExpect, alice.id.String(), test.httpStatus, test.payload)
+	} else {
+		// mint a PO NFT
+		test := struct {
```
Comment (vimukthi-git) [inline testworld/nft_test.go]: same here

File: testworld/nft_test.go [hunk 5]
```diff
@@ -8,6 +8,8 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/gavv/httpexpect"
```
Comment (vimukthi-git) [inline testworld/nft_test.go]: remove ?

File: nft/ethereum_payment_obligation_test.go [hunk 6]
```diff
@@ -3,10 +3,16 @@
...
+	"context"
+	"fmt"
 	"math/big"
 	"testing"
 	"time"
 
+	"github.com/centrifuge/go-centrifuge/config/configstore"
+	"github.com/centrifuge/go-centrifuge/contextutil"
+	"github.com/centrifuge/go-centrifuge/identity"
+
```
Comment (vedhavyas) [inline nft/ethereum_payment_obligation_test.go]: spacing

File: nft/handler.go [hunk 7]
```diff
@@ -64,6 +64,40 @@ func (g grpcHandler) MintNFT(ctx context.Context, request *nftpb.NFTMintRequest)
...
+	}
+	poRegistry := cfg.GetContractAddress(config.PaymentObligation)
+
+	mintReq := &nftpb.NFTMintRequest{
+		Identifier:                request.Identifier,
+		DepositAddress:            request.DepositAddress,
+		RegistryAddress:           poRegistry.Hex(),
+		ProofFields:               proofFields,
+		GrantNftAccess:            true,
+		SubmitNftOwnerAccessProof: true,
```
Comment (vedhavyas) [inline nft/handler.go]: can you merge `GrantNftAccess` and `SubmitNftOwnerAccessProof` to `GrantNftAccess` alone? These two must always be set to true.

Comment (mikiquantum) [reply nft/handler.go]: I rather do that as a separate PR, as there is logic change, functional tests ...

**Pull request merged**